### ISSUE TITLE
cppstd in conanfile.txt

### ISSUE THREAD

**Issue by ghost** [conan-io/conan#3058]
As I understand correctly, either there is currently no way to declare that `cppstd` settings of packages must be respected when building the final application using `conanfile.txt` or there is a bug.

Here's the output I get:

```
/bin/sh -c 'mkdir -p Build && cd Build && conan install .. --settings build_type=Debug && cmake -DCMAKE_BUILD_TYPE=Debug ..'
----------Building project:[ FakeDarkPlacesServer - Debug ]----------
PROJECT: Installing /home/lyberta/Stuff/C++/ftz/QCVM/Samples/conanfile.txt
Requirements
    ftzGeneral/Latest@Lyberta/Latest from 'ftz'
    ftzQCVM/Latest@Lyberta/Latest from local cache
    ftzSerialization/Latest@Lyberta/Latest from local cache
Packages
    ftzGeneral/Latest@Lyberta/Latest:408505463172a58479365902c19aafb19e2e813c
    ftzQCVM/Latest@Lyberta/Latest:1063e11ad7f011a2d25660498b7bea315f0367dd
    ftzSerialization/Latest@Lyberta/Latest:6027bd02ed245c339032344c16cd9864a40d0895
ftzSerialization/Latest@Lyberta/Latest: Already installed!
ftzGeneral/Latest@Lyberta/Latest: Already installed!
ftzQCVM/Latest@Lyberta/Latest: Already installed!
PROJECT: Generator cmake created conanbuildinfo.cmake
PROJECT: Generator txt created conanbuildinfo.txt
PROJECT: Generated conaninfo.txt
-- The CXX compiler identification is GNU 8.1.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Current conanbuildinfo.cmake directory: /home/lyberta/Stuff/C++/ftz/QCVM/Samples/Build
-- Conan: Compiler GCC>=5, checking major version 8
-- Conan: Checking correct version: 8
-- Conan: Using cmake targets configuration
-- Library ftzqcvm found /home/lyberta/.conan/data/ftzQCVM/Latest/Lyberta/Latest/package/1063e11ad7f011a2d25660498b7bea315f0367dd/lib/libftzqcvm.a
-- Library ftzgeneral found /home/lyberta/.conan/data/ftzGeneral/Latest/Lyberta/Latest/package/408505463172a58479365902c19aafb19e2e813c/lib/libftzgeneral.a
-- Library pthread not found in package, might be system one
-- Library ftzserialization found /home/lyberta/.conan/data/ftzSerialization/Latest/Lyberta/Latest/package/6027bd02ed245c339032344c16cd9864a40d0895/lib/libftzserialization.a
-- Library stdc++fs not found in package, might be system one
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Configuring done
-- Generating done
-- Build files have been written to: /home/lyberta/Stuff/C++/ftz/QCVM/Samples/Build
====0 errors, 0 warnings====
```

I expected to see either `Conan setting CXX_FLAGS flags:` or `Conan setting CPP STANDARD:` after `Conan: Adjusting language standard` but there is none. The app doesn't build unless I manually specify flags in CMake.

Here's the extract from `conaninfo.txt`:
```
[full_settings]
    arch=x86_64
    arch_build=x86_64
    build_type=Debug
    compiler=gcc
    compiler.libcxx=libstdc++11
    compiler.version=8
    cppstd=20
    os=Linux
    os_build=Linux
```

**Comment by lasote** [comment]
The `cppstd` is managed by the `CMake()` build helper. So building manually with `cmake` you loose that capability.
You would need to call "conan build ." command instead.

**Comment by Lyberta** [comment]
`conan build` requires `conanfile.py` which I think is too much for most users.

**Comment by lasote** [comment]
Good point. We have an open discussion about limiting what are the build helpers doing "dynamically" and reflect in a generated file (probably different from the generator one), this way you will have everything available calling CMake in the command line. It is our roadmap, but still not possible to do it.

**Comment by memsharded** [comment]
The new ``CMakeToolchain`` generates information to be used by CMake, by simple ``conanfile.txt`` consumers too: https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html

This can be considered implemented (from Conan 1.40 CMakeToolchain is going to be used more toroughly, even the GettingStarted in the docs has changed to use it), so closing this issue.
